design: 매칭 대기 화면 UI 구현

tnemnorivnelee opened the pull request:
# 🔗 관련 이슈

Resolves: #11 

## 📝 작업 내용

- 매칭 대기 화면 화면 퍼블리싱 작업 진행

## 📷 스크린샷 (UI 작업인 경우)

### Android

|           After           | 
| :------------------------: |
| <img src="https://github.com/user-attachments/assets/e2eecafc-e0dd-457e-8723-f057bb478c09" width="373" height="874" /> | 


### iOS

|           After           |
| :------------------------: |
| <img src="https://github.com/user-attachments/assets/5042642d-1b53-4e3d-8dea-fcc6863c5046" width="373" height="874" /> |


## 💬 리뷰어에게

- 뒤로가기가 포함된 헤더 부분을 공통 컴포넌트로 분리했습니다.
- 버튼을 공통 컴포넌트로 분리했습니다.

## ✅ 기본 체크리스트

- [x] 코드가 정상적으로 실행되나요? (`yarn start` 빌드 에러 없음)
- [x] **(중요) 안드로이드/iOS 시뮬레이터에서 각각 확인했나요?**
- [x] 불필요한 `console.log`나 주석을 제거했나요?
- [x] `yarn format`을 실행하여 코드 스타일을 정렬했나요?
- [x] `yarn lint`를 실행하여 린트 에러가 없나요?

## 🏗️ 구조 체크리스트 (Self-Check)

> 💡 아래 항목을 위반하면 코드 리뷰어에게 반려될 수 있습니다!

### 파일 및 폴더

- [x] **1개 기능 전용 컴포넌트**는 `features/` 안에 두었나요? (❌ `shared/ui` 금지)
- [x] **재사용 컴포넌트**만 `shared/ui`에 두었나요?
- [x] **파일이 3개 이상**일 때만 폴더로 묶었나요? (불필요한 `index.ts` 방지)

### 상태 관리 & 네이밍

- [x] **서버 데이터**는 React Query, **UI 상태**는 Zustand를 사용했나요?
- [x] **Boolean 변수**는 `is`, `has`, `should`로 시작하나요?

---

🚀 **모든 체크리스트를 확인한 후 PR을 생성하세요!**

tnemnorivnelee committed:
feat: 매칭 내역 화면 구현 및 공통 UI 리팩토링

- 매칭 내역 화면(MatchHistoryScreen) 추가 및 라우팅 설정
- 공통 헤더 컴포넌트(StackHeader) 구현
- ChatAvatarGroup을 공통 컴포넌트(AvatarGroup)로 리팩토링
- Shared Theme Spacing 토큰 추가 (18px)
- 프로필 화면에 매칭 내역 진입을 위한 임시 버튼 추가

tnemnorivnelee committed:
refactor: contentContainerStyle에서 불필요한 parseInt 제거

- theme.spacing 파싱 대신 직접 숫자 값 사용
- useTheme 훅 제거 (더 이상 컴포넌트에서 theme 미사용)

tnemnorivnelee committed:
refactor: Typography 하드코딩을 theme 토큰으로 변경

- DateText: font-size 12px → theme.typography.fontSize.xs
- TitleText: font-size 18px → theme.typography.fontSize.l
- TitleText: font-weight 600 → theme.typography.fontWeight.semiBold
- 디자인 토큰 일관성 유지를 위한 리팩토링

tnemnorivnelee committed:
refactor: Separator height를 theme.spacing으로 통일

- 하드코딩된 12px를 theme.spacing[12]로 변경
- 디자인 토큰 일관성 유지

tnemnorivnelee committed:
refactor: StackHeader 전체를 디자인 토큰으로 통일

- Spacing: Container, BackButton, RightContainer에 theme.spacing 적용
- Typography: Title에 theme.typography 적용
- 모든 하드코딩된 값 제거하여 디자인 토큰 일관성 확보

tnemnorivnelee committed:
refactor: AvatarGroup에 디자인 토큰 적용

- border-radius: 12px → theme.radius[12]
- background-color: #d9d9d9 → theme.colors.semantic.stroke
하드코딩된 스타일을 theme으로 교체하여 디자인 토큰 일관성 확보

tnemnorivnelee committed:
refactor: BackButton padding 제거

tnemnorivnelee committed:
refactor: StackHeader에서 사용하지 않는 right prop 제거

tnemnorivnelee committed:
refactor: 디자인 토큰을 React Native 표준 숫자 형식으로 변경

- spacing, radius, typography.fontSize를 문자열('4px')에서 숫자(4)로 변경
- fontWeight는 문자열 유지 ('400') - RN 표준

tnemnorivnelee committed:
refactor: 디자인 토큰 시스템 개선 및 정리

- Typography를 문자열 형식('20px')으로 변경하여 styled-components 호환성 개선
- Spacing과 Radius는 실용성을 위해 하드코딩 방식 유지
- MatchHistoryScreen contentContainerStyle을 숫자로 변경 (inline style 표준)

변경 사항:
- theme.typography.fontSize: 숫자 → 문자열 ('12px', '20px' 등)
- styled-components: theme.typography.fontSize.xxx}px → xxx} (px 제거)
- spacing/radius: theme 토큰 → 직접 하드코딩
- colors: theme 사용 유지

이제 Typography만 디자인 토큰으로 관리하며,
Spacing/Radius는 간결성을 위해 하드코딩 방식 사용

tnemnorivnelee committed:
refactor: ChatHeader를 StackHeader로 교체

- ChatRoomScreen에서 ChatHeader 대신 공통 StackHeader 사용

tnemnorivnelee committed:
fix: 폰트 충돌 해결 및 입력창 레이아웃 수정

- 'px' 단위 제거
- 웹 초기 높이 문제 해결
- 입력창 줄 높이(line-height)를 명시하여 렌더링 일관성 확보

tnemnorivnelee committed:
fix: 미사용 코드 제거

tnemnorivnelee committed:
feat: 공통 버튼 컴포넌트 추가

tnemnorivnelee committed:
feat: 매칭 대기 화면 UI 하위 컴포넌트 구현 (경로 정보, 참가자 카드)

tnemnorivnelee committed:
feat: 매칭 대기 화면 구현 및 레이아웃 구성

tnemnorivnelee committed:
feat: 매칭 대기 화면 라우팅 연결

tnemnorivnelee committed:
feat: 매칭 대기 화면 임시 라우팅 버튼 추가

tnemnorivnelee committed:
design: Figma 디자인 명세에 맞춘 수치 적용

tnemnorivnelee committed:
design: 공통 Button 컴포넌트 디자인 피그마에 맞게 수정

tnemnorivnelee committed:
feat: Pretendard 폰트 설정

- Pretendard Regular/Medium/SemiBold/Bold 폰트 로드
- getFontFamily 헬퍼 함수 구현 (shared/lib/typography.ts)

tnemnorivnelee committed:
fix: 매칭 대기 화면 하단 버튼 간격 수정

tnemnorivnelee committed:
refactor: BottomContainer 인라인 스타일을 props로 개선

tnemnorivnelee committed:
fix: BottomContainer를 styled props에서 인라인 스타일로 복원

이전 리팩토링이 레이아웃 문제를 야기하여 원래의 인라인 스타일 방식으로 복원

tnemnorivnelee commented:
ios 이미지 추가

tnemnorivnelee committed:
Merge branch 'develop' of https://github.com/Hansei-ro/hanseiro-app into design/matching-waiting-screen

soyeoung1 commented:
4번째 합류중 카드만 간격이 넓게 떨어져 있어서 이것만 수정하면 될 거 같습니다!

tnemnorivnelee committed:
design: ParticipantCard 간격 수정

tnemnorivnelee commented:
> 4번째 합류중 카드만 간격이 넓게 떨어져 있어서 이것만 수정하면 될 거 같습니다!

@soyeoung1 
오 지금 보니까 확실히 간격이 다르네요,,
간격 수정했습니다 확인 부탁드립니다!

![Screenshot_20251222-005608_hansei-ro 1](https://github.com/user-attachments/assets/6f2a5d3b-0ec7-4871-8de3-d53ad4850410)

Yeram9286 commented on src/features/match/components/ParticipantCard.tsx:
```diff
@@ -0,0 +1,189 @@
+        fill="none"
+        stroke={theme.colors.border.joiningOutline}
+        strokeWidth="1.5"
+        strokeDasharray="6, 4"
```
5,3 이나 4,3 정도로 조절해서 점선을 조금 더 짧고 촘촘하게 만들면 좋을거같습니다

tnemnorivnelee replied:
피그마보다 확실히 점선이 더 긴 느낌이네요,,

수정 완료됐습니다!

![Screenshot_20251222-123955_hansei-ro 1](https://github.com/user-attachments/assets/b9d887b9-f138-4e9a-b321-9ddd15d6f8d2)

Yeram9286 commented on src/features/match/screens/MatchingWaitingScreen.tsx:
```diff
@@ -0,0 +1,100 @@
+        </ScrollView>
+
+        <BottomContainer style={{ paddingBottom: insets.bottom + 16 }}>
+          <Button title="현재 인원으로 출발" onPress={() => {}} variant="primary" />
```
폰트 크기를 16px 정도로 줄여주시면 좋을거같습니다!

tnemnorivnelee replied:
선택해주신 영역의 코드에서는 폰트 크기에 해당하는 부분이 "현재 인원으로 출발" 부분만 보이는데 이 부분은 16px 로 설정되어있습니다!

Yeram9286 reviewed (changes requested):
확인부탁드립니다

tnemnorivnelee committed:
design: 합류 중 테두리 점선 간격 수정

tnemnorivnelee committed:
refactor: 매칭 대기 화면 내 공통 Text 컴포넌트 적용 및 Typography 수정

- MatchingWaitingScreen 및 하위 컴포넌트(MatchingRouteInfo, ParticipantStatusSection, ParticipantCard)의 styled(Text)를 공통 Text 컴포넌트로 교체하여 스타일 일관성 확보
- Button, StackHeader 등 공용 UI 컴포넌트 내부 텍스트도 공통 Text 컴포넌트로 통일

tnemnorivnelee committed:
refactor: match 기능 전반에 공통 Text 컴포넌트 전면 적용

- features/match 내 모든 화면 및 컴포넌트의 텍스트 렌더링 방식을 공통 Text 컴포넌트로 통합
- styled-components 기반의 타이포그래피 스타일 정의 제거
- MatchHistoryScreen 등의 리스트 아이템 레이아웃 구조 개선 (gap 활용)
- 일관된 폰트 스타일 및 lineHeight 적용을 위한 코드베이스 정리
